src/main/java/picard/sam/util/SamComparison.java

```diff
@@ -0,0 +1,520 @@
+        for (final String attribute : attributes) {
+            ret = compareValues(samReadGroupRecord1.getAttribute(attribute), samReadGroupRecord2.getAttribute(attribute),
+                    attribute + " for read group " + samReadGroupRecord1.getReadGroupId()) && ret;
+        }
+        return ret;
+    }
+
+    private boolean compareSequenceDictionaries(final SAMFileHeader h1, final SAMFileHeader h2) {
+        final List<SAMSequenceRecord> s1 = h1.getSequenceDictionary().getSequences();
+        final List<SAMSequenceRecord> s2 = h2.getSequenceDictionary().getSequences();
+        if (s1.size() != s2.size()) {
+            reportDifference(s1.size(), s2.size(), "Length of sequence dictionaries");
+            return false;
+        }
+        boolean ret = true;
+        for (int i = 0; i < s1.size(); ++i) {
+            ret = compareSequenceRecord(s1.get(i), s2.get(i), i + 1) && ret;
+        }
+        return ret;
+    }
+
+    private boolean compareSequenceRecord(final SAMSequenceRecord sequenceRecord1, final SAMSequenceRecord sequenceRecord2, final int which) {
+        if (!sequenceRecord1.getSequenceName().equals(sequenceRecord2.getSequenceName())) {
+            reportDifference(sequenceRecord1.getSequenceName(), sequenceRecord2.getSequenceName(),
+                    "Name of sequence record " + which);
+            return false;
+        }
+        boolean ret = compareValues(sequenceRecord1.getSequenceLength(), sequenceRecord2.getSequenceLength(), "Length of sequence " +
+                sequenceRecord1.getSequenceName());
+        ret = compareValues(sequenceRecord1.getSpecies(), sequenceRecord2.getSpecies(), "Species of sequence " +
+                sequenceRecord1.getSequenceName()) && ret;
+        ret = compareValues(sequenceRecord1.getAssembly(), sequenceRecord2.getAssembly(), "Assembly of sequence " +
+                sequenceRecord1.getSequenceName()) && ret;
+        ret = compareValues(sequenceRecord1.getAttribute("M5"), sequenceRecord2.getAttribute("M5"), "MD5 of sequence " +
+                sequenceRecord1.getSequenceName()) && ret;
+        ret = compareValues(sequenceRecord1.getAttribute("UR"), sequenceRecord2.getAttribute("UR"), "URI of sequence " +
+                sequenceRecord1.getSequenceName()) && ret;
+        return ret;
+    }
+
+    private <T> boolean compareValues(final T v1, final T v2, final String label) {
+        boolean eq = Objects.equals(v1, v2);
+        if (eq) {
+            return true;
+        } else {
+            reportDifference(v1, v2, label);
+            return false;
+        }
+    }
+
+    private void reportDifference(final String s1, final String s2, final String label) {
+        System.out.println(label + " differs.");
+        System.out.println("File 1: " + s1);
+        System.out.println("File 2: " + s2);
+    }
+
+    private void reportDifference(Object o1, Object o2, final String label) {
+        if (o1 == null) {
```

Comment:
You can eliminate these `if`s by using `String.valueOf()` below. Using `String.valueOf()` where `toString()` is used is a nice approach in general, in that it works on objects and scalars alike.